agent-inject/agent/container_env.go

```diff
@@ -59,5 +59,18 @@ func (a *Agent) ContainerEnvVars(init bool) ([]corev1.EnvVar, error) {
 		})
 	}
 
+	// Add IRSA AWS Env variables for vault containers
+	if a.Vault.AuthType == "aws" {
+		envMap := a.getAwsEnvsFromContainer(a.Pod)
+		if len(envMap) == 0 || len(envMap) >= 2 {
+			for k, v := range envMap {
+				envs = append(envs, corev1.EnvVar{
+					Name:  k,
+					Value: v,
+				})
+			}
+		}
```

Comment:
I don't think we need to check the length of the envMap before iterating here?

```suggestion
		for k, v := range envMap {
			envs = append(envs, corev1.EnvVar{
				Name:  k,
				Value: v,
			})
		}
```

Reply:
Yes, it's not needed thr. I'll remove this.